sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -71,7 +72,12 @@ class QueryExecution(
 
   lazy val analyzed: LogicalPlan = executePhase(QueryPlanningTracker.ANALYSIS) {
     // We can't clone `logical` here, which will reset the `_analyzed` flag.
-    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker) transform {
+      // SPARK-35378: Eagerly execute LeafRunnableCommand so that query command with CTE
+      case r: LeafRunnableCommand =>
```

Comment:
we should run all `Command`s, not just `LeafRunnableCommand`

Reply:
OK